### PR TITLE
[YUNIKORN-1274] Remove FailedScheduling pod events from predicate logic

### PR DESCRIPTION
**Pull request by craigcondit**
### What is this PR for?

Removes a confusing warning emitted as Pod event when predicates don't match a pod with a node. As this is a normal, expected occurrence, these warnings are confusing.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1274

### How should this be tested?
No functional changes, but e2e tests have been updated to use the REST API to retrieve these events.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: test/e2e/predicates/predicates_test.go [hunk 1]
```diff
@@ -251,31 +243,25 @@ var _ = Describe("Predicates", func() {
...
-		}
+		By("Verify the YuniKorn request failed scheduling")
+
+		podErr = restClient.WaitForAllocationLog("default", "root."+ns, initPod.ObjectMeta.Labels["applicationId"], podName, 60)
+		Ω(podErr).NotTo(HaveOccurred())
+		log, podErr := restClient.GetAllocationLog("default", "root."+ns, initPod.ObjectMeta.Labels["applicationId"], podName)
+		Ω(podErr).NotTo(HaveOccurred())
+		Ω(len(log)).NotTo(BeZero(), "Log can't be empty")
+
+		Ω(log).Should(ContainElement(HaveKeyWithValue("message", MatchRegexp(".*didn't match Pod's node affinity"))),
```
Comment (manirajv06) [inline test/e2e/predicates/predicates_test.go:263]: Can we check by matching with earlier message? I am assuming allocation log contains same message.

File: test/e2e/predicates/predicates_test.go [hunk 2]
```diff
@@ -410,31 +396,25 @@ var _ = Describe("Predicates", func() {
...
-				Ω(event.Reason).Should(Equal("FailedScheduling"), "Event reason mismatch")
-				framework.Logf("Yunikorn Event Message: %s", event.Message)
-				Ω(event.Message).Should(MatchRegexp("predicate is not satisfied.*taint.*"), "Event message mismatch")
-			}
-		}
+		log, err := restClient.GetAllocationLog("default", "root."+ns, initPod.ObjectMeta.Labels["applicationId"], podNameNoTolerations)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(len(log)).NotTo(BeZero(), "Log can't be empty")
+
+		Ω(log).Should(ContainElement(HaveKeyWithValue("message", MatchRegexp(".*taint.*"))),
```
Comment (manirajv06) [inline test/e2e/predicates/predicates_test.go:416]: Please see my earlier comment. Just depending only on ".*taint.*" may mislead us sometimes.

File: test/e2e/predicates/predicates_test.go [hunk 3]
```diff
@@ -1064,31 +1044,25 @@ var _ = Describe("Predicates", func() {
...
-				Ω(event.Reason).Should(Equal("FailedScheduling"), "Event reason mismatch")
-				framework.Logf("Yunikorn Event Message: %s", event.Message)
-				Ω(event.Message).Should(MatchRegexp("predicate is not satisfied.*free ports.*"), "Event message mismatch")
-			}
-		}
+		log, err := restClient.GetAllocationLog("default", "root."+anotherNS, initPod.ObjectMeta.Labels["applicationId"], labelPodName2)
+		Ω(err).NotTo(HaveOccurred())
+		Ω(len(log)).NotTo(BeZero(), "Log can't be empty")
+
+		Ω(log).Should(ContainElement(HaveKeyWithValue("message", MatchRegexp(".*free ports.*"))),
```
Comment (manirajv06) [inline test/e2e/predicates/predicates_test.go:1064]: Like commented above

File: pkg/plugin/predicates/predicate_manager.go [hunk 4]
```diff
@@ -75,12 +74,7 @@ func (p *predicateManagerImpl) predicatesReserve(pod *v1.Pod, node *framework.No
 func (p *predicateManagerImpl) predicatesAllocate(pod *v1.Pod, node *framework.NodeInfo) (plugin string, error error) {
 	ctx := context.Background()
 	state := framework.NewCycleState()
-	plugin, err := p.podFitsNode(ctx, state, *p.allocationPreFilters, *p.allocationFilters, pod, node)
-	if err != nil {
-		events.GetRecorder().Eventf(pod.DeepCopy(), nil, v1.EventTypeWarning,
-			"FailedScheduling", "FailedScheduling", "predicate is not satisfied, error: %s", err.Error())
-	}
-	return plugin, err
+	return p.podFitsNode(ctx, state, *p.allocationPreFilters, *p.allocationFilters, pod, node)
```
Comment (manirajv06) [inline pkg/plugin/predicates/predicate_manager.go:77]: Should we log this as debug statement?

File: test/e2e/basic_scheduling/basic_scheduling_test.go [hunk 5]
```diff
@@ -96,7 +96,6 @@ var _ = ginkgo.Describe("", func() {
 		gomega.Ω(allocations["partition"]).NotTo(gomega.BeNil())
 		gomega.Ω(allocations["uuid"]).NotTo(gomega.BeNil())
 		gomega.Ω(allocations["applicationId"]).To(gomega.Equal(sleepRespPod.ObjectMeta.Labels["applicationId"]))
-		gomega.Ω(allocations["queueName"]).To(gomega.ContainSubstring(sleepRespPod.ObjectMeta.Namespace))
```
Comment (manirajv06) [inline test/e2e/basic_scheduling/basic_scheduling_test.go:99]: Is this related to the jira? or some other general cleanup?

File: test/e2e/predicates/predicates_test.go [hunk 6]
```diff
@@ -78,20 +77,19 @@ func runTestPod(k *k8s.KubeCtl, conf k8s.SleepPodConfig) *v1.Pod {
...
 	var kClient k8s.KubeCtl
-	//var restClient yunikorn.RClient
+	var restClient yunikorn.RClient
 	var err error
 	var ns, anotherNS string
 	var nodeList *v1.NodeList
 	const LABELVALUE = "testing-label-value"
-	const WARNING = "Warning"
-	//var sleepRespPod *v1.Pod
+	// var sleepRespPod *v1.Pod
```
Comment (manirajv06) [inline test/e2e/predicates/predicates_test.go:85]: can be removed

**Review by manirajv06 — COMMENTED** [review]
Overall Looks good. Minor nits.

File: test/e2e/basic_scheduling/basic_scheduling_test.go (hunk 5 above)
Comment (wilfred-s) [reply test/e2e/basic_scheduling/basic_scheduling_test.go:99]: This is a follow up on YUNIKORN-1269 in which we removed the queueName from object and REST

**Review by wilfred-s — APPROVED** [review]
LGTM
please get approval from @manirajv06 before committing

File: test/e2e/predicates/predicates_test.go (hunk 1 above)
Comment (craigcondit) [reply test/e2e/predicates/predicates_test.go:263]: These match against the message the K8s predicates generate. We purposely keep them generic so as to be more flexible when new K8s versions may change the wording slightly. There's not really a risk of false positives, as only messages from predicates are logged to the allocation log.

File: test/e2e/predicates/predicates_test.go (hunk 2 above)
Comment (craigcondit) [reply test/e2e/predicates/predicates_test.go:416]: Per previous comment, we specifically keep these generic to make supporting a wider range of K8s releases earlier. What we match here is the same as the previous e2e tests, which have been working fine.

File: test/e2e/predicates/predicates_test.go (hunk 3 above)
Comment (craigcondit) [reply test/e2e/predicates/predicates_test.go:1064]: Same response as above.

File: pkg/plugin/predicates/predicate_manager.go (hunk 4 above)
Comment (craigcondit) [reply pkg/plugin/predicates/predicate_manager.go:77]: We don't bother to log this for reservations, I don't really see a need to log here either. This can quickly get overwhelming as this can happen multiple times per second and is of little value.

File: test/e2e/predicates/predicates_test.go (hunk 6 above)
Comment (craigcondit) [reply test/e2e/predicates/predicates_test.go:85]: Will fix during commit.

**Review by manirajv06 — APPROVED** [review]
Ok. As long as there won't be any false positives in message matching, we are good.
